services/hanke-service/src/test/kotlin/fi/hel/haitaton/hanke/logging/DisclosureLogServiceTest.kt

```diff
@@ -91,340 +99,507 @@ internal class DisclosureLogServiceTest {
+                            customerWithContacts = customerWithoutContacts,
+                            contractorWithContacts = customerWithoutContacts,
+                        ),
+                    hankeTunnus = hankeTunnus,
+                )
+            val capturedLogs = slot<Collection<AuditLogEntry>>()
+            every { auditLogService.createAll(capture(capturedLogs)) } returns mutableListOf()
 
-        disclosureLogService.saveDisclosureLogsForApplication(application, userId)
+            disclosureLogService.saveDisclosureLogsForApplication(application, userId)
 
-        verify { auditLogService wasNot Called }
+            val expectedLogs =
+                listOf(HAKIJA, TYON_SUORITTAJA).map {
+                    AuditLogEntryFactory.createReadEntryForCustomer(
+                        application.id!!,
+                        customerWithoutContacts.customer,
+                        it
+                    )
+                }
+            assertThat(capturedLogs.captured).hasSameElementsAs(expectedLogs)
+            verify(exactly = 1) { auditLogService.createAll(any()) }
+        }
+
+        @Test
+        fun `saveDisclosureLogsForApplication with contacts logs contacts`() {
+            val applicationId = 41L
+            val cableReportApplication = ApplicationFactory.createCableReportApplicationData()
+            val firstContact = cableReportApplication.customerWithContacts.contacts[0]
+            val secondContact = cableReportApplication.contractorWithContacts.contacts[0]
+            val application =
+                ApplicationFactory.createApplication(
+                    id = applicationId,
+                    applicationData = cableReportApplication,
+                    hankeTunnus = hankeTunnus
+                )
+            val capturedLogs = slot<Collection<AuditLogEntry>>()
+            every { auditLogService.createAll(capture(capturedLogs)) } returns mutableListOf()
+
+            disclosureLogService.saveDisclosureLogsForApplication(application, userId)
+
+            assertThat(capturedLogs.captured)
+                .containsAtLeast(
+                    AuditLogEntryFactory.createReadEntryForContact(
+                        applicationId,
+                        firstContact,
+                        HAKIJA
+                    ),
+                    AuditLogEntryFactory.createReadEntryForContact(
+                        applicationId,
+                        secondContact,
+                        TYON_SUORITTAJA
+                    ),
+                )
+            verify(exactly = 1) { auditLogService.createAll(any()) }
+        }
     }
 
-    @Test
-    fun `saveDisclosureLogsForApplication doesn't save entries for blank contacts`() {
-        val customerWithoutContacts =
+    @Nested
+    inner class SaveDisclosureLogsForHakemusResponse {
+        @Test
+        fun `saveDisclosureLogsForHakemusResponse with company customers and no contacts does nothing`() {
```

Comment:
When using a nested class for the method, there's no reason to repeat the method name in every test.

Reply:
Removed unnecessary method name text.